examples/combined_ODNP.py

```diff
@@ -0,0 +1,275 @@
+from numpy import *
+from pyspecdata import *
+import os
+import sys
+import SpinCore_pp
+import time
+from Instruments import power_control
+from datetime import datetime
+from SpinCore_pp.verifyParams import verifyParams
+from SpinCore_pp.ppg import run_spin_echo
+# do the same with the inversion recovery
+from SpinCore_pp.power_helper import gen_powerlist
+from pyspecdata.file_saving.hdf_save_dict_to_group import hdf_save_dict_to_group
+import h5py
+# {{{ Combined ODNP
+# {{{ experimental parameters
+# {{{ these need to change for each sample
+output_name = '150mM_TEMPOL_DNP'
+IR_postproc = 'spincore_IR_v1'
+Ep_postproc = 'ODNP_v3'
+adcOffset = 26
+carrierFreq_MHz = 14.897196
+nScans = 1
+nEchoes = 1
+# all times in microseconds
+# note that acq_time_ms is always milliseconds
+p90_us = 4.464
+repetition_us = 0.5e6
+FIR_rd = 0.5e6
+vd_list =r_[2.1e3,1.12e4,2.23e4,3.3e4,4.4e4,5.6e4,6.7e4,7.8e4,8.9e4,1e5] 
+max_power = 4 #W
+power_steps = 14
+threedown = True
+# }}}
+#{{{Power settings
+dB_settings = gen_powerlist(max_power,power_steps+1, three_down=threedown)
+T1_powers_dB = gen_powerlist(max_power, 5, three_down=False)
+T1_node_names = ['FIR_%ddBm'%j for j in T1_powers_dB]
+print("dB_settings",dB_settings)
+print("correspond to powers in Watts",10**(dB_settings/10.-3))
+print("T1_powers_dB",T1_powers_dB)
+print("correspond to powers in Watts",10**(T1_powers_dB/10.-3))
+myinput = input("Look ok?")
+if myinput.lower().startswith('n'):
+    raise ValueError("you said no!!!")
+#}}}
+powers = 1e-3*10**(dB_settings/10.)
+fl = figlist_var()
+save_file=True
+SW_kHz = 3.9 
+acq_time_ms = 1024. # ms
+nPoints = int(acq_time_ms*SW_kHz+0.5)
+acq_time_ms = nPoints/SW_kHz
+deblank_us = 1.0
+tau_us = 3500
+pad_us = 0
+tx_phases = r_[0.0,90.0,180.0,270.0]
+amplitude = 1.0
+deadtime_us = 10.0
+date = datetime.now().strftime('%y%m%d')
+#{{{ enhancement curve pulse prog
+#}}}
+#{{{IR ppg
+def run_scans_IR(vd_list, node_name, power_idx, nScans = 1, rd = FIR_rd, power_on = False, vd_data=None):
+    # (JF delete on reading): convert this to a real docstring!!
+    # nScans is number of scans you want
+    # rd is the repetition delay
+    # power_setting is what you want to run power at
+    # vd list is a list of the vd's you want to use
+    # node_name is the name of the node must specify power
```

Comment:
don't forget to move this to its own function as well

Reply:
in ppg/IR.py